fmc/resource_fmc_prefilter_policy.go

```diff
@@ -0,0 +1,254 @@
+				}
+			}
+			`,
+		CreateContext: resourceFmcPrefilterPolicyCreate,
+		ReadContext:   resourceFmcPrefilterPolicyRead,
+		UpdateContext: resourceFmcPrefilterPolicyUpdate,
+		DeleteContext: resourceFmcPrefilterPolicyDelete,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of this resource",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The description of this resource",
+				Default:     " ",
+				StateFunc: func(val interface{}) string {
+					state := val.(string)
+					if val == nil || state == "" || state == " " {
+						return " "
+					}
+					return state
+				},
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					// Fix for bug in the FMC API which returns " " for empty description
+					if (new == " " && old == "") || (old == " " && new == "") {
+						return true
+					}
+					return old == new
+				},
+			},
+			"default_action": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						//"log_end": {
+						//	Type:        schema.TypeBool,
+						//	Optional: true,
+						//	Description: "Log end",
+						//},
+						"log_begin": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Description: "Log begin",
+						},
+						"send_events_to_fmc": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Description: "Send events to FMC",
+						},
+						"action": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Action. Should be BLOCK_TUNNELS or ANALYZE_TUNNELS",
+							ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+								v := strings.ToUpper(val.(string))
+								allowedValues := []string{"BLOCK_TUNNELS", "ANALYZE_TUNNELS"}
+								for _, allowed := range allowedValues {
+									if v == allowed {
+										return
+									}
+								}
+								errs = append(errs, fmt.Errorf("%q must be in %v, got: %q", key, allowedValues, v))
+								return
+							},
+						},
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+				Description: "Source interface for this resource",
+			},
+		},
+	}
+}
+
+func resourceFmcPrefilterPolicyCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*Client)
+	// Warning or errors can be collected in a slice type
+	// var diags diag.Diagnostics
+	var diags diag.Diagnostics
+
+	var defaultAction = PrefilterPolicyDefaultActionInput{}
+	if inputEntries, ok := d.GetOk("default_action"); ok {
+		entry := inputEntries.([]interface{})[0].(map[string]interface{})
+
+		defaultAction = PrefilterPolicyDefaultActionInput{
+			LogBegin: entry["log_begin"].(bool),
+			//LogEnd:          entry["log_end"].(bool),
```

Comment:
LogEnd seems to be commented out throughout. If this is intended, all good. Might as well remove it if not supported.

Reply:
This was to kept by engineer for future versions. Have removed in this push.